Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift

```diff
@@ -0,0 +1,191 @@
+      .priority(.overrideRequire).isActive = true
+  }
+  
+  open func configure<Message>(
+    _ messageActionButtons: [MessageReactionButtonComponent],
+    message: Message,
+    currentUserId: String,
+    reactionProvider: ReactionProvider,
+    onMessageActionTap: ((MessageReactionButtonComponent?, Message, (() -> Void)?) -> Void)?
+  ) where Message : ManagedMessageViewModel {
+    
+    self.reactionProvider = reactionProvider
+    
+    let reactions = message.messageActions
+      .filter { $0.sourceType == "reaction" }
+      .map { (reaction: $0.value, userID: $0.pubnubUserId) }
+    
+    updateWith(reactions: reactions.map { $0.reaction } )
+    
+    reactionButtons.forEach { button in
+      button.isSelected = reactions.contains(where: { $0.userID == currentUserId && $0.reaction == button.reaction })
+    }
+    
+    reactionButtons.forEach { button in
+      button.didTap({ [weak message] button in
+        guard let message = message else { return }
+        
+        button?.isEnabled = false
+        onMessageActionTap?(button, message) { [weak button] in
+          DispatchQueue.main.async {
+            button?.isEnabled = true
+          }
+        }
+      })
+      .store(in: &button.externalCancellables)
+    }
+  }
+
+  open func configure<Message>(
+    _ message: Message,
+    currentUserId: String,
+    reactionProvider: ReactionProvider,
+    onMessageActionTap: ((MessageReactionButtonComponent?, Message, (() -> Void)?) -> Void)?
+  ) where Message : ManagedMessageViewModel {
+    configure(
+      reactionButtons,
+      message: message,
+      currentUserId: currentUserId,
+      reactionProvider: reactionProvider,
+      onMessageActionTap: onMessageActionTap
+    )
+  }
+  
+  func buttonFor(_ reaction: String) -> MessageReactionButtonComponent {
+    if let button = reactionButtons.first(where: { $0.reaction == reaction }) {
+      return button
+    }
+    
+    let newButton = reactionProvider.makeMessageReactionComponentWith(reaction)
+    reactionButtons.append(newButton)
+    stackViewContainer.stackView.addArrangedSubview(newButton)
+    
+    setNeedsLayout()
+    return newButton
+  }
+}
+
+//MARK: - Private
+private extension MessageReactionListComponent {
+  private func makeCountsDictionary(with reactions: [String]) -> [String: Int] {
+    Dictionary( reactions.map { ($0, 1) }, uniquingKeysWith: +)
+  }
+  
+  private func update(with reaction: String, count: Int) {
+    guard reactionProvider.reactions.contains(reaction) else { return }
+    
+    let button = buttonFor(reaction)
+    button.externalCancellables.forEach { $0.cancel() }
+    button.currentCount = count
+    button.isHidden = false
+  }
+  
+  private func removeUnusedButtons(for reactions: [String]) {
+    var indexes = [Int]()
+    reactionButtons.enumerated().forEach { (index, button) in
+      if !reactions.contains (where: { $0 == button.reaction }) {
+        button.removeFromSuperview()
```

Comment:
Is it required to call `removeFromSuperview` and `removeArrangedSubview`? One of them isn't enough?

Reply:
Unfortunatelu not, as documentation states:
_To prevent the view from appearing on screen after calling the stack’s removeArrangedSubview(_:) method, explicitly remove the view from the subviews array by calling the view’s removeFromSuperview() method, or set the view’s isHidden property to true._